lib/types/zpool.nix

```
@@ -50,6 +53,55 @@
       });
       description = "List of datasets to define";
     };
+    topology = lib.mkOption {
+      type =
+        let
+          vdev = lib.types.submodule ({ name, ... }: {
+            options = {
+              mode = lib.mkOption {
+                type = lib.types.enum modeOptions;
+                default = "";
+                description = "mode of the zfs vdev";
+              };
+              members = lib.mkOption {
+                type = lib.types.listOf lib.types.str;
+                description = "Members of the vdev";
+              };
+            };
+          });
+          parent = config;
+        in
+        lib.types.nullOr
+          (lib.types.submodule
+            ({ config, name, ... }: {
+              options = {
+                type = lib.mkOption {
+                  type = lib.types.enum [ "zfs_topology" ];
+                  default = "zfs_topology";
+                  internal = true;
+                  description = "Type";
+                };
+                # zfs disk types
+                vdev = lib.mkOption {
+                  type = lib.types.listOf vdev;
+                  default = [ ];
+                  description = "A list of storage vdevs";
```

some example would be nice here:

```suggestion
                  description = "A list of storage vdevs";
                  example = [{
                   mode = "mirror";
                   members = [ "x" "y" ];
                  }];
```